src/compiler/commandLineParser.ts

```diff
@@ -76,10 +76,11 @@ namespace ts {
                 "amd": ModuleKind.AMD,
                 "system": ModuleKind.System,
                 "umd": ModuleKind.UMD,
+                "es6": ModuleKind.ES6,

```

Comment:
you do not need that. just leave target as `es6` and allow for additional `--module` flag. today we make it an error to use `--target es6` with `--module`; relaxing this should be sufficient.